be/src/vec/exec/volap_scan_node.cpp

```diff
@@ -257,19 +266,12 @@ Status VOlapScanNode::open(RuntimeState* state) {
             ready = runtime_filter->await();
         }
         if (ready) {
-            std::list<ExprContext*> expr_context;
-            RETURN_IF_ERROR(runtime_filter->get_push_expr_ctxs(&expr_context));
+            std::vector<VExpr*> vexprs;
+            RETURN_IF_ERROR(runtime_filter->get_push_expr_ctxs(&vexprs));
             _runtime_filter_ctxs[i].apply_mark = true;
             _runtime_filter_ctxs[i].runtimefilter = runtime_filter;
 
-            for (auto ctx : expr_context) {
-                ctx->prepare(state, row_desc());
-                ctx->open(state);
-                int index = _conjunct_ctxs.size();
-                _conjunct_ctxs.push_back(ctx);
-                // it's safe to store address from a fix-resized vector
-                _conjunctid_to_runtime_filter_ctxs[index] = &_runtime_filter_ctxs[i];
-            }
+            RETURN_IF_ERROR(_append_rf_into_conjuncts(state, vexprs));
```

Comment:
maybe we should after collect all vexprs from `_runtime_filter_descs` to do `_append_rf_into_conjuncts`